framework/src/controls/PIDTransientControl.C

```diff
@@ -145,7 +145,7 @@ PIDTransientControl::execute()
     _integral += delta * _dt;
     value += _Kint * _integral + _Kpro * delta;
     if (_dt > 0)
-      value += _Kder * delta / _dt;
+      value += _Kder * (delta - _old_delta) / _dt;
```

Comment:
old delta is reset to 0 when the time step fails (see above)
you ll want to add additional logic to prevent using this 0